### PR TITLE
Introduce CassandraSessionConfigurator

### PR DESCRIPTION
**Pull request by Praveen2112**
## Description

`CassandraSessionConfigurator` allows us to isolate various parts of the code related to configuring the Cassandra client and allowing us to have a dedicated config object.module for each component like SSL related configuration. Apart from this we have also introduced a another configuration property `cassandra.security` which should be set to `PASSWORD` if we are using password based authentication for cassandra.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
## Cassandra
* Set `cassandra.security=PASSWORD` along with `cassandra.username` and `cassandra.password` for  password based authentication in cassandra . ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/ssl/TestCassandraSslConfig.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static io.airlift.configuration.testing.ConfigAssertions.assertFullMapping;
+import static io.airlift.configuration.testing.ConfigAssertions.assertRecordedDefaults;
+import static io.airlift.configuration.testing.ConfigAssertions.recordDefaults;
+
+public class TestCassandraSslConfig
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/ssl/TestCassandraSslConfig.java:28]: https://trino.io/docs/current/develop/tests.html
> Test classes should be defined as package-private and final.

Same for `TestCassandraPasswordConfig` class.

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/ssl/TestCassandraSslConfig.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{
+    @Test
+    public void testDefaults()
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/ssl/TestCassandraSslConfig.java:31]: https://trino.io/docs/current/develop/tests.html
> Test methods should be defined as package-private.

Same for `testExplicitPropertyMappings` method and `TestCassandraPasswordConfig` class.

File: plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/ssl/CassandraSslConfig.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+package io.trino.plugin.cassandra.ssl;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.configuration.validation.FileExists;
+
+import java.io.File;
+import java.util.Optional;
+
+public class CassandraSslConfig
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/ssl/CassandraSslConfig.java:23]: `Ssl` → `Tls`. Same for other classes.

**Review by ebyhr — APPROVED** [review]

**Comment by Praveen2112** [comment]
@ebyhr Thanks a lot for the review. AC

File: plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/tls/CassandraSslConfig.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/tls/CassandraSslConfig.java:1]: > CassandraSslConfig.java

Did you forget renaming the file? Same for `TestCassandraSslConfig.java`.

Comment (Praveen2112) [reply plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/tls/CassandraSslConfig.java]: Oops it was missed. Now renamed the file

File: plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientConfig.java [hunk 5]
```diff
@@ -55,6 +53,12 @@
 })
 public class CassandraClientConfig
 {
+    public enum CassandraAuthenticationType
+    {
+        NONE,
+        PASSWORD;
```
Comment (marcinsbd) [inline plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientConfig.java:59]: nit: should we add  ` /**/;` int the end of enum declaration as the pseudo enum option to minimize diff in the future? i.e.
```
NONE,
PASSWORD,
/**/;
```

**Review by marcinsbd — APPROVED** [review]

File: plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientConfig.java (hunk 5 above)
Comment (ebyhr) [reply plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientConfig.java:59]: `;` after `/**/` is redundant. IntelliJ will report it as unnecessary semicolon.

**Comment by Praveen2112** [comment]
@mosabua Can you please take a look at the docs changes ?

File: docs/src/main/sphinx/connector/cassandra.md [hunk 6]
```diff
@@ -49,6 +49,7 @@ The following configuration properties are available:
...
 | `cassandra.allow-drop-table`     | Enables {doc}`/sql/drop-table` operations. Defaults to `false`.                                                                                                                                                                                                                                                                                                                                                             |
+| `cassandra.security`             | Should be set to `PASSWORD` for password based authentication. Defaults to `NONE`.                                                                                                                                                                                                                                                                                                                                          |
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/cassandra.md:52]: Maybe use

```
Configure authentication to Cassandra. Defaults to `NONE`. Set to `PASSWORD` for basic authentication, and configure `cassandra.username` and `cassandra.password`.
```

Also maybe we should redo the two following ones to mention that `cassandra.security=PASSWORD` is required?  

See my following suggestion.

File: docs/src/main/sphinx/connector/cassandra.md [hunk 7]
```diff
@@ -49,6 +49,7 @@ The following configuration properties are available:
...
 | `cassandra.username`             | Username used for authentication to the Cassandra cluster. This is a global setting used for all connections, regardless of the user connected to Trino.                                                                                                                                                                                                                                                                    |
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/cassandra.md:53]:
```suggestion
| `cassandra.username`             | Username for basic authentication to the Cassandra cluster. Requires `cassandra.security=PASSWORD`. |
```

File: docs/src/main/sphinx/connector/cassandra.md [hunk 8]
```diff
@@ -49,6 +49,7 @@ The following configuration properties are available:
...
 | `cassandra.password`             | Password used for authentication to the Cassandra cluster. This is a global setting used for all connections, regardless of the user connected to Trino.                                                                                                                                                                                                                                                                    |
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/cassandra.md:54]:
```suggestion
| `cassandra.password`             | Password for basic authentication to the Cassandra cluster. Requires `cassandra.security=PASSWORD`.                                                                                                                                                                                                                                                                    |
```

**Review by mosabua — COMMENTED** [review]
As requested @Praveen2112  .. did not look at the code myself though.

**Review by mosabua — APPROVED** [review]
Good to go

**Pull request merged**